### PR TITLE
Bpfctl refactoring

### LINKED ISSUE #412: Decompose bpfctl main.go
This main is getting quite large. It might be helpful for future readers and for maintenance to compose some of the logic into discreet functions.

See `bpfctl/src/main.rs`

### PR DESCRIPTION
**Pull request by ykaliuta**
Factor out functionality to own functions

Fixes #412

### CODE REVIEW

**Review by astoycos — COMMENTED** [review]
Pretty much there!! I like what you did with the ListTable and loadCommands implementation but I think we could could eliminate that extra trait to make things simplier!

**Comment by astoycos** [comment]
also `cargo fmt` should fix the immediate failure I'm seeing in CI

File: bpfctl/src/main.rs
Comment (ykaliuta) [reply bpfctl/src/main.rs]: Thanks! Next try :)

File: bpfctl/src/main.rs
Comment (astoycos) [inline bpfctl/src/main.rs:367]:
```suggestion
        // Common fields
        let common = match self {
            Commands::LoadFromFile(l) => {
                LoadRequestCommon{
                    id: &l.id;
                    section_name: &l.section_name;
                    global: &l.global;
                    command: &l.command;
                    location: Some(load_request_common::Location::File(l.path.clone()));
                }
            }
            Commands::LoadFromImage(l) => {
                LoadRequestCommon{
                    id: &l.id;
                    section_name: &l.section_name;
                    global: &l.global;
                    command: &l.command;
                    location: {
                        let image_pull_policy: ImagePullPolicy = l
                            .image_pull_policy
                            .as_str()
                            .try_into()
                            .expect("invalid image pull policy");
                        let (username, password) = match l.registry_auth.clone() {
                            Some(a) => {
                                let auth_raw = general_purpose::STANDARD_NO_PAD.decode(a)?;
                                let auth_string = String::from_utf8(auth_raw)?;
                                let (username, password) = auth_string.split(':').next_tuple().unwrap();

                                (username.to_owned(), password.to_owned())
                            }
                            None => ("".to_owned(), "".to_owned()),
                        };

                        Some(load_request_common::Location::Image(BytecodeImage {
                            url: l.image_url.clone(),
                            image_pull_policy: image_pull_policy as i32,
                            username,
                            password,
                        }))
                    };
                }
            }
            _ => bail!("Unknown command"),
        };

```

File: bpfctl/src/main.rs
Comment (astoycos) [inline bpfctl/src/main.rs:381]:
```suggestion
        if let Some(global) = global {
            for g in global.iter() {
                common.global_data.insert(g.name.to_string(), g.value.clone());
            }
        }

        Ok(Some(common))
    }
```

**Review by astoycos — COMMENTED** [review]
Looking good, thanks for all the hard work just one small var nit :) 

Also if you could make an follow up issue for adding unit tests to all this functionality that would be really great, thanks!

File: bpfctl/src/main.rs
Comment (ykaliuta) [reply bpfctl/src/main.rs]: Hmmm, I would not say I'm fan of the proposal (if I understand it correctly). I pushed an updated version. There is now code duplication for common fields (.to_string() ...) and still fetching `global` from different types require a variable.

**Comment by anfredette** [comment]
> I just want @anfredette To take a final pass since he was just in this code

Looking now...

**Review by anfredette — COMMENTED** [review]
LGTM.  Thanks for doing this @ykaliuta!

I'll let @astoycos approve it after the signing is done.

**Comment by ykaliuta** [comment]
Thanks a lot for the review. Hopefully the final version

**Review by astoycos — APPROVED** [review]
LGTM

**Pull request merged**
